src/schedule/Schedule.ts

```diff
@@ -133,41 +139,41 @@ export class Schedule extends LogEmitter {
 
   /**
    * Stops a scheduled job and removes it from the schedule (but not from the database).
-   * Does not check whether a job exists.
-   *
-   * If not provided with a job name, all scheduled jobs are canceled.
+   * Does nothing if no job with the given name exists.
    *
    * @param name the job to cancel
    */
-  public cancel(name?: string): void {
-    if (name) {
-      this.stop(name);
-      this.logger.debug('cancel', { name });
-      delete this.jobs[name];
-      return;
-    }
+  public cancelJob(name: string): void {
+    this.stopJob(name);
+    this.logger.debug('cancel', { name });
+    delete this.jobs[name];
+  }
 
+  /**
+   * Stops all scheduled jobs and removes them from the schedule (but not from the database).
+   */
+  public cancel(): void {
     this.stop();
     this.logger.debug('cancel all jobs');
     this.jobs = {};
   }
 
   /**
-   * Cancels a scheduled job and removes it from the schedule and the database.
-   * Does not check whether a job exists.
-   *
-   * If not provided with a job name, all scheduled jobs are removed.
+   * Stops a scheduled job and removes it from the schedule and the database.
+   * Does nothing if no job with the given name exists.
    *
    * @param name the job to remove
    */
-  public async remove(name?: string): Promise<void> {
-    if (name) {
-      this.cancel(name);
-      this.logger.debug('remove', { name });
-      await getJobRepository().delete({ name });
-      return;
-    }
+  public async removeJob(name: string): Promise<void> {
+    this.cancelJob(name);
+    this.logger.debug('remove', { name });
+    await getJobRepository().delete({ name }); // does this not fail if the job doesn't exist?
```

Comment:
TODO: verify (e.g. with a test) that this doesn't fail and remove comment

Reply:
Ups, forgot about that comment ^^
Will do